File: examples/mpi/p/procname.py
#!/usr/bin/env python3
################################################################
################################################################
####
#### This program file is part of the book and course
#### "Parallel Computing"
#### by Victor Eijkhout, copyright 2013-2023
####
#### procname.py : get processor name
####
################################################################
################################################################

##codesnippet procnamep
from mpi4py import MPI
procname = MPI.Get_processor_name()
##codesnippet end
comm = MPI.COMM_WORLD
nprocs = comm.Get_size()
procno = comm.Get_rank()

print( f"Process {procno} runs on {procname}" )

File: examples/mpi/mpl/procname.cxx
/* %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
   %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
   %%%%
   %%%% This program file is part of the book and course
   %%%% "Parallel Computing"
   %%%% by Victor Eijkhout, copyright 2020-2026
   %%%%
   %%%% procname.cxx : processor name
   %%%%
   %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
   %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
*/

#include <iostream>
using std::cout;
#include <string>
using std::string;
#include <sstream>
using std::stringstream;
#include <print>
using std::println;

#include <mpl/mpl.hpp>

int main(int argc,char **argv) {

  const mpl::communicator &comm_world = mpl::environment::comm_world();
  int nprocs,procno;
  // compute communicator rank and size
  nprocs = comm_world.size();
  //codesnippet mplprocname
  procno = comm_world.rank();
  string procname =
    mpl::environment::processor_name();
  std::println
    ( "[{}] running on: {}",
      procno,procname );
  //codesnippet end

  return EXIT_SUCCESS;

}
